### PR TITLE
feat(library): Introduce Hugo 0.122 as library

### PR DESCRIPTION
**Pull request by razvand**
Introduce Hugo 0.122 image (ELF Loader) for bincompat runs.

Add `Kraftfile` to build for QMEU and Firecracker on x86_64 using initrd for volume support. The current build uses embedded initrd to build a single image consisting of both the kernel and the initial ramdisk.

Add GitHub workflow to build and deploy the image. Add success badge in `README.md`.

### CODE REVIEW

**Comment by nderjung** [comment]
Included is a mistake, please include a _new_ workflow, and not rename an existing.

File: README.md [hunk 1]
```diff
@@ -43,3 +43,4 @@ kraft pkg ls --apps --update
 | [![](https://github.com/unikraft/catalog/actions/workflows/library-redis7.0.yaml/badge.svg)](https://github.com/unikraft/catalog/actions/workflows/library-redis7.0.yaml) | [`unikraft.org/redis:7.0`](library/redis/7.0) |
 | [![](https://github.com/unikraft/catalog/actions/workflows/library-lua5.4.yaml/badge.svg)](https://github.com/unikraft/catalog/actions/workflows/library-lua5.4.yaml) | [`unikraft.org/lua:5.4`](library/lua/5.4) |
 | [![](https://github.com/unikraft/catalog/actions/workflows/library-node18.yaml/badge.svg)](https://github.com/unikraft/catalog/actions/workflows/library-node18.yaml) | [`unikraft.org/node:18`](library/node/18) |
+| [![](https://github.com/unikraft/catalog/actions/workflows/library-hugo0.122.yaml/badge.svg)](https://github.com/unikraft/catalog/actions/workflows/library-hugo0.122.yaml) | [`unikraft.org/hugo:0.122`](library/hugo/0.122) |
```
Comment (nderjung) [inline README.md:46]: Let's keep entries here alphabetical.

**Review by StefanJum — COMMENTED** [review]
This breaks on the initrd mounting:
```
[    0.102585] ERR:  [libvfscore] Failed to mount initrd0: error -1
[    0.103746] ERR:  [libukboot] Init function at 0x1490f0 returned error -1
```

File: library/hugo/0.122/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+To run this image, [install Unikraft's companion command-line toolchain `kraft`](https://unikraft.org/docs/cli) and then you can run:
+
+```console
+kraft run -p 1313:1313 unikraft.org/hugo:0.122
+```
+
+Query the server using:
+
+```console
+curl localhost:1313:1313
```
Comment (Mihnea0Firoiu) [inline library/hugo/0.122/README.md:14]:
```suggestion
curl localhost:1313
```

Comment (Mihnea0Firoiu) [reply library/hugo/0.122/README.md:14]: Other than this, for me it works!

**Pull request merged**